src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -38,10 +37,8 @@ internal class WorkloadListCommand : CommandBase
             _machineReadableOption = result.ValueForOption<bool>(WorkloadListCommandParser.MachineReadableOption);
             _verbosity = result.ValueForOption<VerbosityOptions>(WorkloadListCommandParser.VerbosityOption);
 
-            _sdkVersion = string.IsNullOrEmpty(result.ValueForOption<string>(WorkloadListCommandParser.VersionOption)) ?
-                new ReleaseVersion(version ?? Product.Version) :
-                new ReleaseVersion(result.ValueForOption<string>(WorkloadListCommandParser.VersionOption));
             var dotnetPath = Path.GetDirectoryName(Environment.ProcessPath);
+            _sdkVersion = WorkloadInstallCommand.GetValidatedSdkVersion(result.ValueForOption<string>(WorkloadUpdateCommandParser.VersionOption), version, dotnetPath);
```

Comment:
nit: If the method is static, there is no harm reaching to a different class. But for the code organization point of view maybe create a separate class, or make it an extension method to sdkverion. Or even create a SDKVersion class if you want to do very OO style